src/qilib/configuration_helper/adapters/m4i_instrument_adapter.py

```diff
@@ -0,0 +1,42 @@
+"""Quantum Inspire library
+
+Copyright 2019 QuTech Delft
+
+qilib is available under the [MIT open-source license](https://opensource.org/licenses/MIT):
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+persons to whom the Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+from typing import Optional
+
+import qcodes
+
+from qilib.configuration_helper.adapters import CommonInstrumentAdapter
+from qilib.utils import PythonJsonStructure
+
+
+class M4iInstrumentAdapter(CommonInstrumentAdapter):
+    """ Adapter for the Spectrum M4i digitizer."""
+
+    def __init__(self, address: str, instrument_name: Optional[str] = None) -> None:
+        super().__init__(address, instrument_name)
+        from qcodes.instrument_drivers.Spectrum.M4i import M4i
```

Comment:
```suggestion
        from qcodes_contrib_drivers.drivers.Spectrum.M4i import M4i
```
And also update the `setup.py` to install `qcodes_contrib_drivers`

Reply:
I also don't like the import in the function, but else when calling the adapters an `ImportError` is raised when the M4i driver is not installed. This is because of;

https://github.com/QCoDeS/Qcodes/blob/270bd7e1e488607304fed91f2e41a5e2c3773295/qcodes/instrument_drivers/Spectrum/pyspcm.py#L62

or 

https://github.com/QCoDeS/Qcodes/blob/270bd7e1e488607304fed91f2e41a5e2c3773295/qcodes/instrument_drivers/Spectrum/pyspcm.py#L163

Reply:
Did update the `qcodes_contrib_drivers.drivers.Spectrum.M4i`.